src/components/Dashboard.tsx

```diff
@@ -33,44 +33,82 @@ const Dashboard: React.FC = () => {
 
   usePageTitle(t('PAGE_TITLE_HOME'))
 
+  const dashboardCategories = templatesByCategory.filter(({ templateCategory: { uiLocation } }) =>
+    uiLocation.includes(UiLocation.Dashboard)
+  )
+
   return (
     <div id="dashboard">
       <Header as="h2" content={t('MENU_ITEM_DASHBOARD')} />
-      {templatesByCategory
-        .filter(({ templateCategory: { uiLocation } }) => uiLocation.includes(UiLocation.Dashboard))
-        .map(({ templates, templateCategory: { icon: categoryIcon, title: categoryTitle } }) => (
-          <div key={categoryTitle} className="template-category">
-            <div className="title">
-              {categoryIcon && <Icon size="large" color="grey" name={categoryIcon} />}
-              <Header as="h4">{categoryTitle}</Header>
-            </div>
-            <div className="templates">
-              {templates.map((template) => (
-                <TemplateComponent key={template.code} template={template} />
-              ))}
-            </div>
-          </div>
+      {dashboardCategories.map(({ templates, templateCategory: { icon, title, code } }) => (
+        <CategoryComponent
+          key={code}
+          templates={templates}
+          icon={icon as SemanticICONS}
+          title={title}
+        />
+      ))}
+    </div>
+  )
+}
+
+const CategoryComponent: React.FC<{
+  templates: TemplateInList[]
+  icon: SemanticICONS
+  title: string
+}> = ({ templates, icon, title }) => {
+  const [readyToDisplay, setReadyToDisplay] = useState(
+    templates.some((t) => t.dashboardRestrictions === null)
+  )
+
+  return (
+    <div key={title} className={`template-category${readyToDisplay ? '' : ' hidden-element'}`}>
+      <div className="title">
+        {icon && <Icon size="large" color="grey" name={icon} />}
+        <Header as="h4">{title}</Header>
+      </div>
+      <div className="templates">
+        {templates.map((template) => (
+          <TemplateComponent key={template.code} template={template} setReady={setReadyToDisplay} />
```

Comment:
So the tricky thing here was that we also want to hide categories if there are no templates within it that should appear. However, normally the Category is rendered before all the filters are loaded/checked. Which would have resulted in Categories (and even templates) appearing on the Dashboard for a few seconds until the filters loaded, then disappearing.

So I've added some additional logic here:
- Categories (and templates) that have no restrictions appear as normal (immediately)
- If any of the templates have restrictions, then they remain hidden until at least one filter returns a positive result.